superset/db_engine_specs/ocient.py

```diff
@@ -114,3 +117,22 @@ def fetch_data(cls, cursor, lim=None):
         if len(data) != 0 and type(data[0]).__name__ == 'Row':
             data = [list(map(_verify_datatype, r)) for r in data]
         return data
+    
+    @classmethod
+    def get_cancel_query_id(cls, cursor: Any, query: Query) -> Optional[str]:
+        # Store the cursor locally so that we can retrieve it if we need to cancel
+        cursor_cache[query.id] = cursor
+
+        # Return a non-None object
+        # If none is returned, superset will not call cancel_query
+        return ''
+
+    @classmethod
+    def cancel_query(cls, cursor: Any, query: Query, cancel_query_id: str) -> bool:
+        # If query.id is not in the cache, it must have been cancelled by a different call to cancel_query
+        if query.id in cursor_cache:
+            try:
+                cursor.execute(f'CANCEL {cursor_cache[query.id].query_id}')
+            except Exception:
+                return False
```

Comment:
Maybe log this error instead of swallowing

Reply:
Error now propagates to superset